src/SFA.DAS.FindApprenticeshipTraining/Application/TrainingCourses/LocationHelper.cs

```diff
@@ -18,13 +19,23 @@ public LocationHelper (ILocationApiClient<LocationApiConfiguration> locationApiC
 
         public async Task<GetLocationsListItem> GetLocationInformation(string location)
         {
-            if (!string.IsNullOrEmpty(location) && location.Split(",").Length == 2)
+
+            if (string.IsNullOrEmpty(location))
+            {
+                return null;
+            }
+
+            if (location.Split(",").Length == 2)
             {
                 var locationInformation = location.Split(",");
                 var locationName = locationInformation.First().Trim();
                 var authorityName = locationInformation.Last().Trim();
                 return await _locationApiClient.Get<GetLocationsListItem>(new GetLocationByLocationAndAuthorityName(locationName, authorityName));
             }   
+            if (Regex.IsMatch(location, @"^[A-Za-z]{1,2}\d[A-Za-z\d]?\s*\d[A-Za-z]{2}$"))
```

Comment:
either pull that regex into a variable that describes the check, or pull the whole check into a method describing what its doing